pkg/controller/nodeipam/ipam/cidrset/cidr_set_test.go

```diff
@@ -341,6 +343,77 @@ func TestCIDRSet_AllocationOccupied(t *testing.T) {
 	}
 }
 
+func TestDoubleOccupyRelease(t *testing.T) {
+	clusterCIDRStr := "10.42.0.0/16"
+	operations := []struct {
+		cidrStr     string
+		occupy      bool
+		numOccupied int
+	}{
+		{
+			cidrStr:     "10.42.5.0/24",
+			occupy:      true,
+			numOccupied: 1,
+		},
+		{
+			cidrStr:     "10.42.9.0/24",
+			occupy:      true,
+			numOccupied: 2,
+		},
+		{
+			cidrStr:     "10.42.8.0/22",
+			occupy:      true,
+			numOccupied: 5,
+		},
+		{
+			cidrStr:     "10.42.9.0/24",
+			occupy:      true,
+			numOccupied: 5,
+		},
+		{
+			cidrStr:     "10.42.9.0/24",
+			occupy:      false,
+			numOccupied: 4,
+		},
+		{
```

Comment:
release an previous deallocated subnet, should not decrease the number of occupied